HIP/hip-904-frictionless-airdrop.md

```diff
@@ -266,194 +454,312 @@ Note: the only change to this API is in the specification, which now allows for
- * All cancellations must succeed for the transaction to succeed.
+ * Token cancel airdrop<br/>
+ * Remove one or more pending airdrops from state on behalf of the sender(s)
+ * for each airdrop.<br/>
+ * 
+ * Each pending airdrop canceled SHALL be removed from state and SHALL NOT be available to claim.
+ * 
+ * Each cancellation SHALL be represented in the transaction body and SHALL NOT be restated
+ * in the record file.
+ * All cancellations MUST succeed for this transaction to succeed.
  */
 message TokenCancelAirdropTransactionBody {
     /**
-    * A list of up to 10 pending airdrops that may be cancelled. If there are
-    * duplicate pending_airdrops, then the transaction will fail.
-    */
-    repeated TokenPendingAirdrop pending_airdrops = 1;
+     * A list of pending airdrop items to cancel.<br/>
+     * This transaction MUST be signed by the account referenced by a `sender_id` for
+     * each entry in this list.
+     * <p>
+     * This list MUST contain between 1 and 10 entries, inclusive.
+     * This list MUST NOT have any duplicate entries.<br/>
+     */
+    repeated PendingAirdrop pending_airdrops = 1;
 }
 ```
 
 #### Claim
 
 ```protobuf
 /**
- * A token cancel will clear a pending transfer from state. 
- * This transfer will be represented in the transaction body and will not need restating in the record file.
- * All claims must succeed for the transaction to succeed.
+ * Token claim airdrop<br/>
+ * Complete one or more pending transfers on behalf of the recipient(s) for each airdrop.<br/> 
+ * The sender MUST have sufficient balance to fulfill the airdrop at the time of claim. If the
+ * sender does not have sufficient balance, the claim SHALL fail.
+ * 
+ * Each pending airdrop successfully claimed SHALL be removed from state and SHALL NOT be available
+ * to claim again.
+ * 
+ * Each claim SHALL be represented in the transaction body and SHALL NOT be restated
+ * in the record file.<br/>
+ * All claims MUST succeed for this transaction to succeed.
  */
 message TokenClaimAirdropTransactionBody {
     /**
-    * The pending airdrop to be accepted. The transaction must be signed by
-    * the `receiverAccountID`'s key as contained in the pending airdrop.
-    * A list of up to 10 pending airdrops that may be claimed. If there are
-    * duplicate pending_airdrops, then the transaction will fail.
-    */
-    repeated TokenPendingAirdrop pending_airdrops = 1;
+     * A list of one or more pending airdrops.<br/>
+     * This transaction MUST be signed by the account referenced in the `receiver_id` for
+     * each entry in this list.
+     * <p>
+     * This list MUST contain between 1 and 10 entries, inclusive.<br/>
+     * This list MUST NOT have any duplicate entries.
+     */
+    repeated PendingAirdrop pending_airdrops = 1;
 }
 ```
 
 #### TransactionRecord
 
 ```protobuf
 message TransactionRecord {
-    
-    ...	
-    
+
```

Comment:
Isn't this naming too general? Probably we can use `AirdropTransactionRecord` to be more specific.